lib/cuda/cudaCorrelator.hpp

```diff
@@ -0,0 +1,68 @@
+#ifndef KOTEKAN_CUDA_CORRELATOR_CUH
+#define KOTEKAN_CUDA_CORRELATOR_CUH
+
+#include "cudaCommand.hpp"
+#include "cudaDeviceInterface.hpp"
+#include "n2k.hpp"
+
+/**
+ * @class cudaCorrelator
+ * @brief cudaCommand for doing an N2 correlation, with Cuda code from Kendrick.
+ *
+ * @author Andre Renard (by kindly telling Dustin Lang what to type)
+ *
+ * An example of this stage being used can be found in
+ * `config/tests/verify_cuda_n2k.yaml`.
+ *
+ * A CPU implementation is in `lib/testing/gpuSimulateN2k.hpp`.
+ *
+ * Reads config values:
+ * - num_elements -- number of feeds
+ * - num_local_freq -- frequencies
+ * - samples_per_data_set -- number of time samples per Kotekan block
+ * - sub_integration_ntime -- number of time samples that will be summed
+ *     into the correlation matrix
+ *
+ * Input buffer (gpu):
+ * - gpu_mem_voltage -- size per frame: samples_per_data_set * num_elements * num_local_freq
+ *   -4+4-bit complex voltage inputs
+ *
+ * Output buffer (gpu):
+ * - gpu_mem_correlation_triangle -- size per frame: (samples_per_data_set / sub_integration_ntimes)
+ * * num_elements^2 * 2 * sizeof(int32) -Complex correlation values
```

Comment:
For the headers, please use doxygen tags for the buffers and config parameters.  We haven't made a ton of use of them yet, but the idea is to be able to auto generate documents for each `CommandObject/Stage`, and if nothing else it creates a standardized way of listing these.   See for example: https://github.com/kotekan/kotekan/blob/develop/lib/hsa/hsaBeamformHFBSum.hpp for a command object, and https://github.com/kotekan/kotekan/blob/develop/lib/stages/eigenVis.hpp for a stage.

Reply:
Done!